lib/run_loop/app.rb

```diff
@@ -14,6 +14,20 @@ class App
     # @return [RunLoop::App] A instance of App with a path.
     def initialize(app_bundle_path)
       @path = File.expand_path(app_bundle_path)
+
+      if !App.valid?(app_bundle_path)
```

Comment:
Immediately raise an error if the app bundle is invalid.  This avoids having to check in each instance method.